opm/simulators/wells/WellState.cpp

```diff
@@ -785,6 +787,10 @@ void WellState<Scalar>::initWellStateMSWell(const std::vector<Well>& wells_ecl,
                         if (first_perf > -1) { //-1 indicates that the global id is not on this process
                             segment_pressure[seg] = perf_press[first_perf];
                         }
+                        // The segment_pressure needs to be communicated right away, otherwise the segment_pressure for the
+                        // case segment_perforations[seg].empty() is incorrect, since it uses the pressure of the outlet segment,
+                        // which is only on one process!
+                        segment_pressure[seg] = ws.parallel_info.get().communication().sum(segment_pressure[seg]);
```

Comment:
A global communication inside an if statement. Is `segment_perforations[seg].empty()]` the same on all processes?